tests/test_sync.py

```diff
@@ -56,6 +55,57 @@ def test_synchronize():
             cleanup_remote_repo(owner, repository, result)
 
 
+def test_synchronize_to_existing_pr():
+    """Test synchronization tool (when PR already exists)."""
+
+    # Define your test data here
+    owner = "ansys"
+    repository = "ansys-tools-repo-sync"
+    from_dir = os.path.join(ASSETS_DIRECTORY, "ansys")
+    to_dir = "src/ansys"
+    manifest = os.path.join(ASSETS_DIRECTORY, "manifest.txt")
+
+    # Call the function
+    result = None
+    try:
+        result = synchronize(
+            owner=owner,
+            repository=repository,
+            token=TOKEN,
+            from_dir=from_dir,
+            to_dir=to_dir,
+            include_manifest=manifest,
+            skip_ci=True,
+        )
+
+        # Assertions or validations
+        assert f"https://github.com/ansys/ansys-tools-repo-sync/pull/" in result
+
+        # Call the function again - and check that the PR already exists.
+        result_pr_already_exists = synchronize(
+            owner=owner,
+            repository=repository,
+            token=TOKEN,
+            from_dir=from_dir,
+            to_dir=to_dir,
+            include_manifest=manifest,
+            skip_ci=True,
+        )
+
+        # Verify the PR is the same
+        assert result_pr_already_exists == result
+
+        # Check that the proper modified files have been added
+        list_of_files = ["src/ansys/api/test/v0/hello_world.py", "src/ansys/api/test/v0/test.proto"]
+        assert check_files_in_pr(owner, repository, result, list_of_files)
+
+    except Exception as err:
+        raise err
+    finally:
+        if result:
+            cleanup_remote_repo(owner, repository, result)

```

Comment:
Maybe we should clean up the `result_pr_already_exists` too if possible, just in case `result` is None for some reason. I don't know if `cleanup_remote_repo` will complain about trying to delete the same thing twice... 

```suggestion
        if result:
            cleanup_remote_repo(owner, repository, result)
        if result_pr_already_exists:
            cleanup_remote_repo(owner, result_pr_already_exists, result)
```

Reply:
Hmm good idea... Let me add that

Reply:
Let's test it in the workflow directly

Reply:
FYI - it's best as you originally posted it. Two independent ifs =) that's the one I merged. That way, if they are not the same, for whatever reason, they both get deleted.